src/app-layer.c

```diff
@@ -845,6 +846,7 @@ int AppLayerHandleTCPData(ThreadVars *tv, TcpReassemblyThreadCtx *ra_ctx, Packet
                     ExceptionPolicyApply(
                             p, g_applayerparser_error_policy, PKT_DROP_REASON_APPLAYER_ERROR);
                     AppLayerIncrErrorExcPolicyCounter(tv, f, g_applayerparser_error_policy);
+                    AppLayerIncFlowCounter(tv, f);
```

Comment:
I think the same logic applies for `AppLayerHandleTCPData` as for `AppLayerHandleUdp`

Flow counts is now done as part of protocol detection : we wait for the protocol to be detected to report the flow
It is not a logical part of parsing

Reply:
> Flow counts is now done as part of protocol detection : we wait for the protocol to be detected to report the flow It is not a logical part of parsing

I see. Thank you.

Q: Why doesn't the detection run for every input then? Is it costly?

Context: Once a dcerpc request/response is completely done, a new input in the same direction is given to the parser. This is now parsed and verified in the header and fails here (So, it's a parser error). The probing was never run for this input which led it to directly fail in the parser while it could have easily failed in the detection phase iff detection was run on it using the probing fn.

Reply:
> > Why doesn't the detection run for every input then? Is it costly?
> 
> Why should we ? Once we detected the app-layer protocol, we use it...
> 

It sounds all good to me but then we encounter weird data in the stream that does not belong the protocol that was initially detected. It ideally should be rejected by the protocol parser but then for weird pcaps like the one in QA, it makes error stats go sky high.

> > The probing was never run for this input which led it to directly fail in the parser while it could have easily failed in the detection phase iff detection was run on it using the probing fn.
> 
> In the pcap shared, the probing parser should have been run for this input as it is the first data from the server, right ?
> 
> But yes, we do want to error somehow on parsing for this scenario " Once a dcerpc request/response is completely done, a new input in the same direction is given to the parser."

hmm that's what I'm trying to do.. I'll get back on this after we have reached a conclusion for https://redmine.openinfosecfoundation.org/issues/7238